src/typing_extensions/utils.py

```diff
@@ -0,0 +1,99 @@
+"""
+Protocols and type aliases that are not scheduled for inclusion in typing.
+"""
+
+from os import PathLike
+from typing import AbstractSet, Awaitable, Iterable, Tuple, TypeVar, Union
+from typing_extensions import Protocol, TypeAlias
+
+_T_co = TypeVar("_T_co", covariant=True)
+_T_contra = TypeVar("_T_contra", contravariant=True)
+_KT = TypeVar("_KT")
+_KT_co = TypeVar("_KT_co", covariant=True)
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT_co = TypeVar("_VT_co", covariant=True)
+
+#
+# Protocols for dunder methods
+#
+
+
+class SupportsAnext(Protocol[_T_co]):
+    def __anext__(self) -> Awaitable[_T_co]:
+        ...
+
+
+class SupportsDivMod(Protocol[_T_contra, _T_co]):
```

Comment:
I still think we should drop the divmod protocols:

- They're not widely used: in typeshed we only use them for `divmod` itself.
- They're unintuitive in that divmod() normally returns a pair, but the protocols don't reflect that.